packages/svelte/src/internal/client/dom/blocks/if.js

```diff
@@ -33,35 +33,49 @@ export function if_block(node, fn, elseif = false) {
 	/** @type {UNINITIALIZED | boolean | null} */
 	var condition = UNINITIALIZED;
 
-	var flags = elseif ? EFFECT_TRANSPARENT : 0;
+	var flags = root_index > 0 ? EFFECT_TRANSPARENT : 0;
 
 	var has_branch = false;
 
-	const set_branch = (/** @type {(anchor: Node) => void} */ fn, flag = true) => {
+	const set_branch = (
+		/** @type {(anchor: Node, elseif?: [number,number]) => void} */ fn,
+		flag = true
+	) => {
 		has_branch = true;
 		update_branch(flag, fn);
 	};
 
 	const update_branch = (
 		/** @type {boolean | null} */ new_condition,
-		/** @type {null | ((anchor: Node) => void)} */ fn
+		/** @type {null | ((anchor: Node, elseif?: [number,number]) => void)} */ fn
 	) => {
 		if (condition === (condition = new_condition)) return;
 
 		/** Whether or not there was a hydration mismatch. Needs to be a `let` or else it isn't treeshaken out */
 		let mismatch = false;
 
-		if (hydrating) {
-			const is_else = /** @type {Comment} */ (anchor).data === HYDRATION_START_ELSE;
+		if (hydrating && hydrate_index !== -1) {
+			if (root_index === 0) {
+				const data = /** @type {Comment} */ (anchor).data;
+				if (data === HYDRATION_START) {
+					hydrate_index = 0;
+				} else if (data === HYDRATION_START_ELSE) {
+					hydrate_index = Infinity;
+				} else {
+					hydrate_index = parseInt(data.substring(1));
+				}
+			}
+			const is_else = hydrate_index > root_index;
 
-			if (!!condition === is_else) {
+			if (!!condition === is_else || isNaN(hydrate_index)) {
```

Comment:
We definitely don't want to be using `isNaN` anywhere, it's a major codesmell

Reply:
I didn't known that.
What the alternative to detect bad hydration comment ?

Reply:
Validate it before doing `parseInt`?

Reply:
What about `hydrate_index !== hydrate_index`, which will be true only for NaN ?

Reply:
Yes in this case the indexes will not match.

Here it is rather the case where the comment is not from the Svelte hydration process...

Reply:
`isNaN` has been condemned to hell all over, you can research its history. I'd just rather avoid it and it's caveats where possible.